doc/source/operations/secret-rotation.rst

```diff
@@ -0,0 +1,528 @@
+===============
+Secret Rotation
+===============
+
+General notes
+=============
+
+This guide covers secret rotation in Kayobe and Kolla-Ansible for most services
+in a standard deployment. It does not cover every secret. A full list of
+passwords that have been successfully rotated is available at the bottom of
+this page (:ref:`link<full-password-list>`).
+
+Many of the secrets can simply be deleted from your ``passwords.yml`` and will
+be automatically regenerated with a ``kayobe overcloud service deploy``.
+
+Some secrets require manual input from the operator to change.
+
+Following this process, there may be a few seconds of network downtime for
+running VMs when Neutron is reconfigured when using ML2/OVS.
+
+There will be API downtime for all services. The main reason for the outage is
+that RabbitMQ must be completely stopped to change the secrets it uses. The
+services must all be reconfigured to use the new RabbitMQ cluster. Each service
+will come back once it has been reconfigured. The outage time for each service
+is therefore equal to the time between starting a ``kayobe overcloud service
+deploy``, and that service being reconfigured.
+
+Some secrets currently have to be regenerated by hand. Make sure you use a
+reliable tool and match the formatting (length, character set etc) of the
+existing secret. ``pwgen`` is recommended and used as an example throughout
+this guide. Installation:
+
+.. code:: bash
+
+   sudo apt/dnf install pwgen
+
+
+As of writing, there are three upstream patches in the works to make this
+process easier.
+
+#. A change to Kolla, to automate :ref:`this<kolla-change>` step to change the
+   extended start for the ``nova-api`` container.
+
+   The upstream patch can be found `here
+   <https://review.opendev.org/c/openstack/kolla/+/902057>`__.
+
+   This was previously mitigated with a change to the StackHPC fork of
+   Kolla-Ansible, which has since been reverted due to an unforeseen issue.
```

Comment:
This is just vague enough to not be useful imo ;)
I think it either needs a link to the revert so people can check they're not using it, or just remove this sentence.

Reply:
I like the mystery. All my PRs have rich lore

Reply:
Done